lib/matplotlib/_mathtext.py

```diff
@@ -2294,6 +2301,18 @@ def non_math(self, s, loc, toks):
         self.get_state().font = mpl.rcParams['mathtext.default']
         return [hlist]
 
+    def text(self, s, loc, toks):
+        self.pop_state()
```

Comment:
This won't work with something like `r"$\mathbf{\text{hello, world!}}$"` (popping out of the mathbf puts you in italic mode, and setting back to italic at exit isn't correct either; instead I think you just need to push a new state for the `\text`.

Reply:
Ahh, OK! I'll look into the state things. Almost looked a bit too easy to solve it...